src/Control/Effect/Internal/Error.hs

```diff
@@ -261,6 +303,50 @@ errorToIO m =
   $ m
 {-# INLINE errorToIO #-}
 
+newtype ErrorToIOAsExcC e m a = ErrorToIOAsExcC {
+    unErrorToIOAsExcC ::
+      IntroC '[Catch e, Throw e] '[ErrorIO]
+        (ErrorToErrorIOAsExcC e
+          (ErrorIOToIOC m)) a
+  } deriving ( Functor, Applicative, Monad
+             , Alternative, MonadPlus
+             , MonadFix, MonadFail, MonadIO
+             , MonadThrow, MonadCatch, MonadMask
+             , MonadBase b, MonadBaseControl b
+             )
+    deriving (MonadTrans, MonadTransControl)
+    via CompositionBaseT
+     '[ IntroC '[Catch e, Throw e] '[ErrorIO]
+      , ErrorToErrorIOAsExcC e
+      , ErrorIOToIOC
+      ]
+
+deriving instance (Eff (Embed IO) m, Exception e, C.MonadCatch m) => Carrier (ErrorToIOAsExcC e m)
+
+-- | Runs connected 'Throw' and 'Catch' effects -- i.e. 'Error' --
+-- by treating @e@ as an 'IO' exception.
+--
+-- @'Derivs' ('ErrorToIOAsExcC' e m) = 'Catch' e ': 'Throw' e ': 'Derivs' m@
+--
+-- @'Control.Effect.Primitive.Prims' ('ErrorToIOAsExcC' e m) = 'Control.Effect.Primitive.Prims' m@
```

Comment:
The use of `ErrorIOToIOC` introduces `Optional SomeException` as a primitive effect, so this is wrong. It should be:
```haskell
-- @'Control.Effect.Primitive.Prims' ('ErrorToIOAsExcC' e m) = 'Control.Effect.Optional.Optional' ((->) 'Control.Exception.SomeException') ': 'Control.Effect.Primitive.Prims' m@
```